assets/js/app/Data.js

```diff
@@ -47,9 +99,19 @@ LocusZoom.Data.Requester = function(sources) {
     };
 };
 
-LocusZoom.Data.AssociationSource = function(url) {
-    this.url = url;
-
+LocusZoom.Data.AssociationSource = function(init) {
+    if (LocusZoom.isString(init)) {
+        this.url = init;
+        this.params = {};
+    } else {
+        this.url = init.url;
+        this.params = init.params || {};
+    }
+    if (!this.url) {
+        throw("AssociationSource not initialized with required URL");
+    }
+    
+    var that=this;
```

Comment:
Good suggestion. I had not used that syntax before. Does it work well with all the promise stuff?